webapp/src/js/directives/navigation.js

```diff
@@ -4,22 +4,41 @@ angular.module('inboxDirectives').directive('mmNavigation', function() {
   return {
     restrict: 'E',
     templateUrl: 'templates/directives/filters/navigation.html',
-    controller: function($ngRedux, $scope, GlobalActions, Selectors) {
+    controller: function(
+      $ngRedux,
+      $scope,
+      $state,
+      $stateParams,
+      GlobalActions,
+      Selectors
+    ) {
       'ngInject';
 
       const ctrl = this;
-      const mapStateToTarget = function(state) {
+      const mapStateToTarget = state => {
         return {
           cancelCallback: Selectors.getCancelCallback(state),
           enketoSaving: Selectors.getEnketoSavingStatus(state),
           title: Selectors.getTitle(state)
         };
       };
       const mapDispatchToTarget = function(dispatch) {
-        return GlobalActions(dispatch);
+        const globalActions = GlobalActions(dispatch);
+        return { unsetSelected: globalActions.unsetSelected };
       };
       const unsubscribe = $ngRedux.connect(mapStateToTarget, mapDispatchToTarget)(ctrl);
 
+      /**
+       * Navigate back to the previous view
+       */
+      ctrl.navigateBack = () => {
+        if ($state.current.name === 'contacts.deceased') {
+          $state.go('contacts.detail', { id: $stateParams.id });
+        } else {
+          ctrl.unsetSelected();
```

Comment:
I think removing the additional condition:
```
if ($stateParams.id) {
          $state.go($state.current.name, { id: null });
```
is screwing up tasks.

Reply:
I think it was screwed up by a previous commit. Fixed now.